### DISCUSSION TITLE
How to specify vcpkg package features?

### DISCUSSION THREAD

**Discussion by cursey** [build-cpp/cmkr#40] [Q&A] [answered]
Some vcpkg packages have optional features that you can specify in `vcpkg.json` for example:
```
{
    ...
    "dependencies": [
        {
            "name": "glad",
            "features": ["gl-api-30"]
        }
    ]
}
```

Is there a way to specify features like this in `cmake.toml`?

**Comment by mrexodia** [comment, marked as answer]
Hello,

There is currently no way to do this. I saw this feature of vcpkg but never used it. Any suggestions for the syntax?

```
[vcpkg]
packages = [“glad:gl-api-30”]
```

Or something else?

**Reply by cursey** [reply]
I'm not super familiar with toml so I'm unsure what the ideal syntax for this would be, but it needs to support multiple features for example:
```
{
    ...
    "dependencies": [
        {
            "name": "imgui",
            "features": ["docking-experimental", "freetype", "sdl2-binding", "opengl3-binding"]
        }
    ]
}
```

**Reply by cursey** [reply]
Having given it more thought, maybe just use the same syntax vcpkg uses (which supports multiple features). Something like
```
[vcpkg]
packages = ["glad[gl-api-30]", "imgui[docking-experimental,freetype,sdl2-binding,opengl3-binding]"]
```

**Reply by cursey** [reply]
I went ahead and implemented it. I'll create PR shortly for you to review.
